mlos_bench/mlos_bench/optimizers/mlos_core_optimizer.py

```diff
@@ -104,20 +104,19 @@ def bulk_register(self,
         df_scores = self._adjust_signs_df(
             pd.DataFrame([{} if score is None else score for score in scores]))
 
-        opt_targets = list(self._opt_targets)
         if status is not None:
             # Select only the completed trials, set scores for failed trials to +inf.
             df_status = pd.Series(status)
             # TODO: Be more flexible with values used for failed trials (not just +inf).
             # Issue: https://github.com/microsoft/MLOS/issues/523
-            df_scores.loc[df_status != Status.SUCCEEDED, opt_targets] = float("inf")
+            df_scores[df_status != Status.SUCCEEDED] = float("inf")
```

Comment:
What kind of tests can we put in for this?  For backwards compatibility and forwards bug fix checks?

Reply:
good point - I'll add tests to this PR